yt_dlp/extractor/tvp.py

```diff
@@ -503,6 +506,14 @@ def _parse_video(self, video, with_url=True):
             'thumbnails': ('images', ..., ..., {'url': ('url', {url_or_none})}),
         })
         info_dict['description'] = clean_html(dict_get(video, ('lead', 'description')))
+        if type_ in ('PROGRAMME', 'LIVE'):
+            now = datetime.datetime.now().astimezone(datetime.timezone.utc).timestamp()
+            if parse_iso8601(video.get('since')) > now:
+                info_dict['live_status'] = 'is_upcoming'
+            elif video.get('till') is None or parse_iso8601(video.get('till')) >= now:
+                info_dict['live_status'] = 'is_live'
+            else:
+                info_dict['live_status'] = 'was_live'
```

Comment:
I'm not familiar with the site, but by the looks of things, type `PROGRAMME` are the EPG entries. If we raise on date/don't even match date, is it even required to handle them here? Maybe I missed some other URL format that is also of the `PROGRAMME` type.

Could you also share an example of an upcoming/expired livestream?